subiquity/server/controllers/mirror.py

```diff
@@ -143,13 +163,39 @@ async def apply_autoinstall_config(self, context):
             log.debug("Skipping mirror check since network is not available.")
             return
 
-        try:
-            await self.try_mirror_checking_once()
-        except AptConfigCheckError:
-            log.debug("Retrying in 10 seconds...")
+        # Try each mirror one after another.
+        compatibles = self.model.compatible_primary_candidates()
+        for idx, candidate in enumerate(compatibles):
+            if idx != 0:
+                # Sleep before testing the next candidate..
+                log.debug("Will check next candiate mirror after 10 seconds.")
+                await asyncio.sleep(10)
+            if candidate.uri is None:
+                log.debug("Skipping unresolved country mirror")
```

Comment:
if candidate.uri is ever none for any other reason this log statement will be confusing

Reply:
if it's none for any other reason, this means something has gone very wrong :)
If you think it's a problem, I can add a getter for `uri` and make it `assert self.country_mirror` if uri is None.

Reply:
I'm thinking about my recent experience in Curtin where I believe that no swap was being created, and really it was a swap file problem.
Yes, we should have hit a ValueError before getting this far.
I think it's potentially interesting to store the raw input configuration line with the entry, which should be only a small change and means you would be able to log it here.  WDYT?

Reply:
I'm not completely sure I got what you're saying. 

By raw input configuration, you mean the original yaml autoinstall section?
There is not always an original yaml. For the desktop installer use case, for instance, the two candidates will be the default ones created automatically by `MirrorModel`.

Would you like to see a logline with the autoinstall representation of each candidate we try? That I can do with `candidate.serialize_for_ai()`

Reply:
> By raw input configuration, you mean the original yaml autoinstall section?

Well, one line from it at least, the relevant object from version-2 primary list.

> Would you like to see a logline with the autoinstall representation of each candidate we try? That I can do with candidate.serialize_for_ai()

Works for me, I'm looking to leverage something we have at hand that is just more clear.